command/e2etest/main_test.go

```diff
@@ -0,0 +1,296 @@
+package e2etest
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	tfcore "github.com/hashicorp/terraform/terraform"
+)
+
+var terraformBin string
+
+func TestMain(m *testing.M) {
+	if os.Getenv("GOOS") != "" || os.Getenv("GOARCH") != "" {
```

Comment:
I don't think you need this check. You can't run tests when cross compiling because you won't be on the correct OS to execute them, and if the `-c` flag is being used to ship a binary to another host, the environment variable have no reason to be set, or any effect at runtime.